redwood-widget/src/commonMain/kotlin/app/cash/redwood/widget/MutableListChildren.kt

```diff
@@ -17,12 +17,23 @@ package app.cash.redwood.widget
 
 import kotlin.jvm.JvmOverloads
 
+/**
+ * A [MutableList] that is also a [Widget.Children].
+ *
+ * @param list Optional existing [MutableList] instance to wrap.
+ * @param onUpdate Optional callback invoked when contents change via the [Widget.Children] API.
+ */
 public class MutableListChildren<T : Any>
 @JvmOverloads constructor(
-  public val list: MutableList<T> = mutableListOf(),
-) : Widget.Children<T>, Iterable<T> {
+  private val list: MutableList<T> = mutableListOf(),
+  private val onUpdate: (List<T>) -> Unit = {},
+) : Widget.Children<T>, MutableList<T> by list {
+  /** @param onUpdate Callback invoked when contents change via the [Widget.Children] API. */
+  public constructor(onUpdate: (List<T>) -> Unit) : this(mutableListOf(), onUpdate = onUpdate)
```

Comment:
necessary for parameter order?

Reply:
Primarily for exposing to iOS where things like default argument values (as used in primary constructor) do not work. Although I have not yet tested that it shows up in the header properly.